<launch>
  <arg name="limb" default="right"/>
  <arg name="raw_mode" default="False"/>
  
  <!-- Enable baxter robot -->
  <node name="enable_baxter_robot" pkg="baxter_tools" type="enable_robot.py" output="screen" args="-e"/>
  
  <!-- the semantic description that corresponds to the URDF -->
  <param name="robot_description_semantic" textfile="$(find baxter_moveit_config)/config/baxter.srdf" />
  <!-- Start the kinematics services node -->
  <node name="kinematics_services" pkg="moveit_kinematics_interface" type="kinematics_services" output="screen">
    <rosparam command="load" file="$(find baxter_moveit_config)/config/kinematics.yaml"/>
    <param name="planning_group" value="$(arg limb)_arm"/>
    <param name="ik_dynamic_reconfigure" value="false"/>
    <param name="metrics_enabled" value="False"/>
  </node>

  <!-- Position mode benchmarking node -->
  <node name="repeatability_benchmark" pkg="baxter_benchmark" type="repeatability_benchmark.py" output="screen" required="True">
    <rosparam command="load" file="$(find baxter_benchmark)/config/repeatability.yaml"/>
    <param name="raw_mode" value="$(arg raw_mode)"/>
    <param name="limb" value="$(arg limb)"/>
  </node>

</launch>
